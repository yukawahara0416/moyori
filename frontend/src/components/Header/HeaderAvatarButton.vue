<template>
  <div class="ml-3 text-center">
    <v-menu offset-y>
      <template v-slot:activator="{ on, attrs }">
        <v-btn class="test" icon v-bind="attrs" v-on="on">
          <header-avatar-image :currentUser="currentUser" />
          <v-icon>mdi-chevron-down</v-icon>
        </v-btn>
      </template>

      <header-avatar-list :currentUser="currentUser" />
    </v-menu>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import HeaderAvatarImage from '@/components/Header/HeaderAvatarImage.vue'
import HeaderAvatarList from '@/components/Header/HeaderAvatarList.vue'

export default {
  components: {
    HeaderAvatarImage,
    HeaderAvatarList
  },

  computed: {
    ...mapGetters(['currentUser'])
  }
}
</script>

<style scoped>
button::before {
  background-color: white;
}
</style>
