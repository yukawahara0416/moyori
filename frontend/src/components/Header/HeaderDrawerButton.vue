<template>
  <v-navigation-drawer
    app
    temporary
    right
    class="drawer"
    :value="value"
    @input="input($event)"
  >
    <v-list>
      <header-drawer-pages />

      <v-divider />

      <header-drawer-footer />
    </v-list>
  </v-navigation-drawer>
</template>

<script>
import HeaderDrawerPages from '@/components/Header/HeaderDrawerPages.vue'
import HeaderDrawerFooter from '@/components/Header/HeaderDrawerFooter.vue'

export default {
  props: {
    value: {
      type: Boolean,
      default: () => {
        return false
      },
      required: true
    }
  },

  components: {
    HeaderDrawerPages,
    HeaderDrawerFooter
  },

  methods: {
    input(event) {
      this.$emit('input', event)
    }
  }
}
</script>

<style scoped>
.drawer {
  z-index: 200;
}
</style>
