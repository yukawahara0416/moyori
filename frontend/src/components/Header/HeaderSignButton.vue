<template>
  <v-dialog v-model="dialog" width="600">
    <template v-slot:activator="{ on }">
      <div class="my-2">
        <v-btn
          color="primary"
          depressed
          large
          v-show="headers === null"
          v-on="on"
          @click="dialogOn"
        >
          はじめる
        </v-btn>
      </div>
    </template>

    <sign-container />
  </v-dialog>
</template>

<script>
import { mapGetters } from 'vuex'
import SignContainer from '@/components/Sign/SignContainer.vue'

export default {
  components: {
    SignContainer
  },

  computed: {
    ...mapGetters(['headers', 'dialogSign']),

    dialog: {
      get() {
        return this.dialogSign
      },
      set() {
        this.$store.dispatch('dialogOff')
      }
    }
  },

  methods: {
    dialogOn() {
      this.$store.dispatch('dialogOn', 'dialogSign')
    }
  }
}
</script>
