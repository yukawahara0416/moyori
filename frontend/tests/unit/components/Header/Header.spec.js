import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
// import Vuetify from 'vuetify'
import Component from '@/components/Header/Header.vue'
import HeaderAvatarButton from '@/components/Header/HeaderAvatarButton.vue'
import HeaderSignButton from '@/components/Header/HeaderSignButton.vue'
import HeaderDrawerButton from '@/components/Header/HeaderDrawerButton.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
// localVue.use(Vuetify)

let wrapper
let store
let auth
let $route
// let vuetify

beforeEach(() => {
  auth = {
    getters: {
      isLoggingIn: () => false
    }
  }

  store = new Vuex.Store({
    modules: {
      auth
    }
  })

  $route = {
    name: 'search'
  }

  // vuetify = new Vuetify()
})

describe('with shallowMount wrapper', () => {
  beforeEach(() => {
    wrapper = shallowMount(Component, {
      localVue,
      store,
      // vuetify,
      mocks: {
        $route
      }
    })
  })

  describe('getters', () => {
    it('isLoggingIn', () => {
      expect(wrapper.vm.isLoggingIn).toEqual(store.getters.isLoggingIn)
    })
  })

  describe('computed', () => {
    it('isSearchRoute is true', () => {
      expect(wrapper.vm.isSearchRoute).toBeTruthy()
    })

    it('isSearchRoute is false', () => {
      throw new Error('テスト未作成')
    })
  })

  describe('methods', () => {
    it('openDrawer', () => {
      wrapper.setData({ drawerState: false })

      wrapper.vm.openDrawer()
      expect(wrapper.vm.drawerState).toBeTruthy()
    })
  })

  describe('template', () => {
    it('snapshot', () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})

describe('with mount wrapper', () => {
  beforeEach(() => {
    wrapper = mount(Component, {
      localVue,
      store,
      // vuetify,
      mocks: {
        $route
      },
      stubs: [
        'v-app-bar',
        'v-toolbar',
        'header-title',
        'header-tutorial-button',
        'header-avatar-button',
        'header-sign-button',
        'header-drawer-button'
      ]
    })
  })

  describe('v-on', () => {
    it('openDrawer', () => {
      const openDrawer = jest.fn()

      // wrapper.setMethods({ openDrawer })

      wrapper.find('.v-app-bar__nav-icon').trigger('click')
      expect(openDrawer).toHaveBeenCalled()
    })
  })

  describe('template', () => {
    it('v-if="isLoggingIn"', () => {
      auth.getters.isLoggingIn = () => true

      store = new Vuex.Store({
        modules: {
          auth
        }
      })

      wrapper = shallowMount(Component, {
        localVue,
        store
      })

      expect(wrapper.findAll(HeaderAvatarButton).length).toEqual(1)
      expect(wrapper.findAll(HeaderSignButton).length).toEqual(0)
    })

    it('v-else', () => {
      auth.getters.isLoggingIn = () => false

      store = new Vuex.Store({
        modules: {
          auth
        }
      })

      wrapper = shallowMount(Component, {
        localVue,
        store
      })

      expect(wrapper.findAll(HeaderAvatarButton).length).toEqual(0)
      expect(wrapper.findAll(HeaderSignButton).length).toEqual(1)
    })

    it('HeaderDrawerButton has :drawerState', () => {
      expect(wrapper.find(HeaderDrawerButton).props().value).toEqual(
        wrapper.vm.drawerState
      )
    })

    it('snapshot', () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
