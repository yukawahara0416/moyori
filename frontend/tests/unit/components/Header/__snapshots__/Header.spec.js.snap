// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div>
  <v-app-bar-stub
    app="true"
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    tile="true"
    value="true"
  >
    <v-toolbar-stub
      extensionheight="48"
      flat="true"
      src=""
      tag="header"
      tile="true"
    >
      <header-title-stub />
       
      <header-chip-stub />
       
      <v-spacer-stub />
       
      <header-avatar-stub
        currentuser="[object Object]"
      />
       
      <v-app-bar-nav-icon-stub />
    </v-toolbar-stub>
  </v-app-bar-stub>
   
  <header-drawer-stub />
</div>
`;
